### PR TITLE
feat: [IDP-1953]: Git providers support added for create repo, direct push and register catalog steps

### PR DESCRIPTION
**Pull request by bhavya-sinha**
### Summary
Added Github, gitlab, bitbucket and azure git providers and respective fields for IDP native steps i.e; Create repo, register catalog and direct push steps

Will add coverage as a separate PR.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [x] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by bhavya-sinha** [comment]
trigger messagecheck

**Comment by bhavya-sinha** [comment]
trigger messagecheck

**Review by jenil-harness — APPROVED** [review]
LGTM

**Review by jenil-harness — APPROVED** [review]

File: src/modules/75-idp/components/PipelineSteps/CreateRepoStep/CreateRepoStepEdit.tsx [hunk 1]
```diff
@@ -148,10 +139,27 @@ const CreateRepoStepEdit = (
...
+                  if (connectorSelected !== formik?.values?.spec.connectorType) {
+                    formik?.setFieldValue('spec.connectorRef', undefined)
+                    formik?.setFieldValue('spec.description', '')
+                    formik?.setFieldValue('spec.defaultBranch', '')
+                    formik?.setFieldValue('spec.organization', '')
+                    formik?.setFieldValue('spec.repository', '')
+                    formik?.setFieldValue('spec.project', '')
+                    formik?.setFieldValue('spec.workspace', '')
+                  }
+                  formik?.setFieldValue('spec.connectorType', connectorSelected)
```
Comment (chetanvekariya-harness) [inline src/modules/75-idp/components/PipelineSteps/CreateRepoStep/CreateRepoStepEdit.tsx:156]: nit: We can do this in if condition itself. We do not need to set field value if connector is same as what is there in formik.

File: src/modules/75-idp/components/PipelineSteps/DirectPushStep/DirectPushStepEdit.tsx [hunk 2]
```diff
@@ -19,52 +18,23 @@ import { validate } from '@pipeline/components/PipelineSteps/Steps/StepsValidate
...
-  {
-    initialValues,
-    onUpdate,
-    isNewStep = true,
-    readonly,
-    stepViewType,
-    onChange,
-    allowableTypes
-  }: DirectPushStepEditProps,
+  props: DirectPushStepEditProps,
```
Comment (chetanvekariya-harness) [inline src/modules/75-idp/components/PipelineSteps/DirectPushStep/DirectPushStepEdit.tsx:34]: Are this and other step exactly same/similar ? If yes, we can have common component and use at both places.

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by chetanvekariya-harness — COMMENTED** [review]
Plan for test cases?

File: src/modules/75-idp/components/PipelineSteps/DirectPushStep/DirectPushStepEdit.tsx (hunk 2 above)
Comment (bhavya-sinha) [reply src/modules/75-idp/components/PipelineSteps/DirectPushStep/DirectPushStepEdit.tsx:34]: NO These steps are different.

**Comment by bhavya-sinha** [comment]
> Plan for test cases?

Yes. Will pick up post GA

**Pull request merged**